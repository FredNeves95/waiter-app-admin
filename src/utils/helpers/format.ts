export const formatToBRCurrency = new Intl.NumberFormat('pt-BR', {
  style: 'currency',
  currency: 'BRL'
});
