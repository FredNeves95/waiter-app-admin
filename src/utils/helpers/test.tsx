import 'jsdom-global/register';
import { mount, shallow,   } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import { theme } from '../../styles/theme';
import { ReactNode } from 'react';
import { ActivePageProvider } from '../../contexts/ActivePageContext';
import { ModalContextProvider } from '../../contexts/ModalContext';
import { OrderContextProvider } from '../../contexts/OrderContext';

export function mountWithTheme(children: ReactNode) {
  return mount(
    <ThemeProvider theme={theme}>
      <ModalContextProvider>
        <ActivePageProvider>
          <OrderContextProvider>
            {children}
          </OrderContextProvider>
        </ActivePageProvider>
      </ModalContextProvider>
    </ThemeProvider>);
}

export function shallowWithTheme(children: ReactNode) {
  return shallow(
    <ThemeProvider theme={theme}>
      <ModalContextProvider>
        <ActivePageProvider>
          <OrderContextProvider>
            {children}
          </OrderContextProvider>
        </ActivePageProvider>
      </ModalContextProvider>
    </ThemeProvider>);
}
