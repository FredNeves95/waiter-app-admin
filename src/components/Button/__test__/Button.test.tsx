import {beforeEach, describe, expect, it, vi} from 'vitest';
import {Button, ButtonProps} from '../index';
import { mountWithTheme } from '../../../utils/helpers/test';


describe('<Button/>', () => {
  const handleClick = vi.fn();

  beforeEach(() => {
    vi.clearAllMocks();
  });

  const renderComponent = (props?: ButtonProps) =>
    mountWithTheme(<Button {...props} onClick={handleClick}/>);

  it('Should match the snapshot', () => {
    const buttonPrimary = renderComponent();
    const buttonSecondary = renderComponent({buttonType: 'secondary'});

    expect(buttonPrimary.html()).toMatchSnapshot();
    expect(buttonSecondary.html()).toMatchSnapshot();
  });

  it('Should render button', () => {
    const button = renderComponent();
    expect(button.isEmptyRender()).toBeFalsy();
  });

  it('Should render button with children received from props', () => {
    const button = renderComponent({
      children: <div id='mocked-children'>test</div>
    });

    const child = button.find('#mocked-children');
    expect(child.isEmptyRender()).toBeFalsy();
    expect(child.text()).toEqual('test');
  });

  it('Should execute handleClick function after clicking the button', () => {
    const button = renderComponent();
    expect(handleClick).not.toHaveBeenCalled();
    button.simulate('click');
    expect(handleClick).toHaveBeenCalled();
  });

  it('Should not execute handleClick function after clicking the button when button is disabled', () => {
    const button = renderComponent({disabled: true});
    expect(handleClick).not.toHaveBeenCalled();
    button.simulate('click');
    expect(handleClick).not.toHaveBeenCalled();
  });
});
