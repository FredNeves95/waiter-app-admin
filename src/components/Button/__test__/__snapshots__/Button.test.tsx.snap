// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Button/> > Should match the snapshot 1`] = `"<button type=\\"button\\" button-type=\\"primary\\" class=\\"sc-bcPKhP bikja-d\\"></button>"`;

exports[`<Button/> > Should match the snapshot 2`] = `"<button type=\\"button\\" button-type=\\"secondary\\" class=\\"sc-bcPKhP bZFRJa\\"></button>"`;
